packages/babel-plugin-transform-es2015-parameters/src/rest.js

```diff
@@ -177,6 +177,7 @@ function optimiseIndexGetter(path, argsId, offset) {
     scope.push({ id: temp, kind: "var" });
     path.parentPath.replaceWith(restIndexImpure({
       ARGUMENTS: argsId,
+      OFFSET: t.numericLiteral(offset),
```

Comment:
Good catch, thank you. I've added a commit to address this. Also I tried to optimize the AST to avoid test like `1 < 2`... in the pure path. However, I don't know how to evaluate the node without putting it in a path first because `evaluate()` is an instance method of a path. So I had to put the node into the tree first, then evaluate the path before replacing node. A bit hacky I suppose. If you have any suggestion to improve this, please let me know.